File: CS/ConfigureConnectionStringAtRuntime.Web/DocumentDataStorageProvider.cs
using System;
using System.ComponentModel.Composition;
using DevExpress.Xpf.Printing.Service.Extensions;
using DevExpress.Xpo.DB;

namespace ConfigureConnectionStringAtRuntime.Web {
    [Export(typeof(IDocumentDataStorageProvider))]
    public class DocumentDataStorageProvider : IDocumentDataStorageProvider {
        #region IDocumentStoreProvider Members

        string IDocumentDataStorageProvider.ConnectionString {
            get { return AccessConnectionProvider.GetConnectionString("|DataDirectory|/ReportService.mdb"); }
        }

        TimeSpan IDocumentDataStorageProvider.KeepInterval {
            get { return this.GetDefaultKeepInterval(); }
        }

        int IDocumentDataStorageProvider.BinaryStorageChunkSize {
            get { return this.GetDefaultBinaryStorageChunkSize(); }
        }

        #endregion
    }
}

File: VB/ConfigureConnectionStringAtRuntime.Web/DocumentDataStorageProvider.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpf.Printing.Service.Extensions
Imports DevExpress.Xpo.DB
Imports System.ComponentModel.Composition

Namespace ConfigureConnectionStringAtRuntime.Web
    <Export(GetType(IDocumentDataStorageProvider))> _
    Public Class DocumentDataStorageProvider
        Implements IDocumentDataStorageProvider
#Region "IDocumentStoreProvider Members"

        Public ReadOnly Property ConnectionString() As String Implements IDocumentDataStorageProvider.ConnectionString
            Get
                Return AccessConnectionProvider.GetConnectionString("|DataDirectory|/ReportService.mdb")
            End Get
        End Property

        Public ReadOnly Property KeepInterval() As TimeSpan Implements IDocumentDataStorageProvider.KeepInterval
            Get
                Return Me.GetDefaultKeepInterval()
            End Get
        End Property

        Public ReadOnly Property BinaryStorageChunkSize() As Integer Implements IDocumentDataStorageProvider.BinaryStorageChunkSize
            Get
                Return Me.GetDefaultBinaryStorageChunkSize()
            End Get
        End Property
#End Region
    End Class
End Namespace